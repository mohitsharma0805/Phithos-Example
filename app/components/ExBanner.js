import React from 'react';
import { View, Text, StyleSheet, Platform } from 'react-native';
import DeviceInfo from 'react-native-device-info';
import config from '../config';

const ExBanner = () => {
  const getDeviceType = () => {
    let deviceType = DeviceInfo.getModel();
    if (DeviceInfo.isEmulatorSync) {
      if (Platform.OS === 'ios') {
        deviceType = 'Simutator';
      } else {
        deviceType = 'Emulator';
      }
    }
    return deviceType;
  };

  return (
    <View style={styles.contentView}>
      <Text style={styles.text}>{getDeviceType()}</Text>
    </View>
  );
};

export default ExBanner;

const styles = StyleSheet.create({
  contentView: {
    justifyContent: 'center',
    height: 40,
    width: '100%',
    backgroundColor: config.colors.black,
  },
  text: { marginLeft: 20, color: config.colors.white, fontSize: 17 },
});
