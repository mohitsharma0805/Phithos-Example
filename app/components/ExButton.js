import React from 'react';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';
import config from '../config';

const ExButton = ({ buttonStyle, text, onPress }) => {
  return (
    <TouchableOpacity onPress={onPress} style={[styles.button, buttonStyle]}>
      <Text style={styles.text}>{text}</Text>
    </TouchableOpacity>
  );
};

export default ExButton;

const styles = StyleSheet.create({
  button: {
    justifyContent: 'center',
    alignItems: 'center',
    marginHorizontal: 20,
    borderRadius: 5,
    height: 50,
    backgroundColor: config.colors.appTheme,
  },
  text: { color: config.colors.white, fontSize: 17 },
});
