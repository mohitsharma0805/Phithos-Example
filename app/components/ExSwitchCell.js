import React from 'react';
import { View, StyleSheet, Text, Switch } from 'react-native';
import config from '../config';

const ExSwitchCell = ({ switchValue, onSwitchValueChange }) => {
  const switchValueChanged = (value) => onSwitchValueChange?.(value);
  return (
    <View style={styles.container}>
      <View style={styles.wrapper}>
        <View style={styles.textWrapper}>
          <Text style={styles.title}>Banner Status</Text>
        </View>
        <Switch
          style={styles.switch}
          value={switchValue}
          onValueChange={switchValueChanged}
          trackColor={{
            true: config.colors.appTheme,
            false: null,
          }}
        />
      </View>
    </View>
  );
};

export default ExSwitchCell;

const styles = StyleSheet.create({
  container: { backgroundColor: config.colors.white },
  wrapper: { flexDirection: 'row', alignItems: 'center', height: 60 },
  textWrapper: { flex: 1 },
  title: {
    marginLeft: 16,
    paddingVertical: 2,
    fontSize: 17,
  },
  switch: { marginRight: 20, marginLeft: 16 },
});
