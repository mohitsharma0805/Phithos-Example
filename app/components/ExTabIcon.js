import React from 'react';
import { View, StyleSheet } from 'react-native';
import SimpleLineIcons from 'react-native-vector-icons/SimpleLineIcons';

const ExTabIcon = ({ containerStyle, iconName, iconStyle }) => {
  return (
    <View style={[styles.container, containerStyle]}>
      <SimpleLineIcons name={iconName} style={[styles.icon, iconStyle]} />
    </View>
  );
};

export default ExTabIcon;

const styles = StyleSheet.create({
  container: {
    width: 26,
    height: 26,
    margin: 5,
  },
  icon: { marginTop: 4, fontSize: 22 },
});
