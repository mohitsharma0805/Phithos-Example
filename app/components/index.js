export { default as ExButton } from './ExButton';
export { default as ExTextInput } from './ExTextInput';
export { default as ExBanner } from './ExBanner';
export { default as ExSwitchCell } from './ExSwitchCell';
