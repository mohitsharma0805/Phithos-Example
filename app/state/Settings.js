import { createSlice } from '@reduxjs/toolkit';

const saveBannerData = (state, action) => {
  state.banner = action.payload;
};

// REDUCER
const settingsSlice = createSlice({
  name: 'settings',
  initialState: {
    banner: true,
  },
  reducers: {
    saveBannerStatus: saveBannerData,
  },
});

// ACTIONS
const { saveBannerStatus } = settingsSlice.actions;

// SELECTOR
const selectBannerStatus = ({ settings }) => settings.banner;

const settingsReducer = settingsSlice.reducer;

export { settingsReducer, saveBannerStatus, selectBannerStatus };
