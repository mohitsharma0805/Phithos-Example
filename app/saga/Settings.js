import { put, takeLatest } from 'redux-saga/effects';
import { Settings } from '../state';

export function* updateBannerStatus(action) {
  // call api here
  console.log('action.payload', action.payload);
  yield put(Settings.saveBannerStatus(action.payload.status));
}

export function* watchBannerStatus() {
  yield takeLatest('UPDATE_BANNER_STATUS', updateBannerStatus);
}
