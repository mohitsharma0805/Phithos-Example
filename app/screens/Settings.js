import React from 'react';
import { StyleSheet, SafeAreaView } from 'react-native';
import { useSelector, useDispatch } from 'react-redux';
import { ExBanner, ExSwitchCell } from '../components';
import { selectBannerStatus, saveBannerStatus } from '../state/Settings';

const Settings = () => {
  const dispatch = useDispatch();
  const bannerStatus = useSelector(selectBannerStatus);

  const switchValueChanged = (value) => {
    dispatch(saveBannerStatus(value));
  };

  return (
    <SafeAreaView style={styles.contentView}>
      {bannerStatus && <ExBanner />}
      <ExSwitchCell switchValue={bannerStatus} onSwitchValueChange={switchValueChanged} />
    </SafeAreaView>
  );
};

export default Settings;

const styles = StyleSheet.create({
  contentView: { flex: 1 },
  container: { flex: 1, justifyContent: 'center' },
});
