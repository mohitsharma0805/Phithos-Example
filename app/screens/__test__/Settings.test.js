import React from 'react';
import { shallow } from 'enzyme';
import { SafeAreaView } from 'react-native';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import Settings from '../Settings';
import { ExBanner, ExSwitchCell } from '../../components';

describe('Login', () => {
  const dispatch = jest.fn();
  redux.useDispatch.mockReturnValue(dispatch);
  describe('Rendering with show banner', () => {
    const mockStore = configureStore();
    const store = mockStore({
      banner: true,
    });

    jest.spyOn(redux, 'useSelector').mockImplementationOnce(() => store.getState().banner);

    const wrapper = shallow(<Settings />);
    it('renders the main view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(SafeAreaView).length).toEqual(1);
      expect(wrapper.find(ExSwitchCell).length).toEqual(1);
      expect(wrapper.find(ExBanner).length).toEqual(1);
    });
    it('Simulate TextInput and Button', () => {
      const switchCell = wrapper.find(ExSwitchCell).first();
      switchCell.props().onSwitchValueChange();
      expect(switchCell).toBeTruthy();
    });
  });
  describe('Rendering with hide banner', () => {
    const mockStore = configureStore();
    const store = mockStore({
      banner: false,
    });

    jest.spyOn(redux, 'useSelector').mockImplementationOnce(() => store.getState().banner);

    const wrapper = shallow(<Settings />);
    it('renders the main view', () => {
      expect(wrapper).toMatchSnapshot();
      expect(wrapper.find(SafeAreaView).length).toEqual(1);
      expect(wrapper.find(ExSwitchCell).length).toEqual(1);
      expect(wrapper.find(ExBanner).length).toEqual(0);
    });
    it('Simulate TextInput and Button', () => {
      const switchCell = wrapper.find(ExSwitchCell).first();
      switchCell.props().onSwitchValueChange();
      expect(switchCell).toBeTruthy();
    });
  });
});
