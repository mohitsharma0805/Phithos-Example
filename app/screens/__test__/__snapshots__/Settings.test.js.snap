// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Rendering with hide banner renders the main view 1`] = `ShallowWrapper {}`;

exports[`Login Rendering with show banner renders the main view 1`] = `ShallowWrapper {}`;

exports[`Login UI test cases renders the main view 1`] = `ShallowWrapper {}`;
